web/src/todo_widget.ts

```diff
@@ -136,9 +131,11 @@ export class TaskData {
                 return undefined;
             },
 
-            inbound: (sender_id, data) => {
+            inbound: (sender_id: number | string, raw_data: unknown): void => {
                 // All readers may add tasks. For legacy reasons, the
                 // inbound idx is called key in the event.
+
+                const data = new_task_inbound_data_schema.parse(raw_data);
                 const idx = data.key;
                 const task = data.task;
                 const desc = data.desc;
```

Comment:
Since we are parsing this `unknown` data correctly using `zod` should we remove the following unnecessary type checks?

Reply:
Yes. Done. Thanks!